example-widget-mui/src/UploadImagePage/UploadImage.tsx

```diff
@@ -0,0 +1,263 @@
+
+  return (
+    <>
+      <NavigationBar title="Upload image" />
+
+      <Box m={1}>
+        <MuiCapabilitiesGuard
+          capabilities={[
+            WidgetEventCapability.forRoomEvent(
+              EventDirection.Receive,
+              ROOM_EVENT_UPLOADED_IMAGE,
+            ),
+            WidgetEventCapability.forStateEvent(
+              EventDirection.Receive,
+              STATE_EVENT_POWER_LEVELS,
+            ),
+            WidgetEventCapability.forStateEvent(
+              EventDirection.Receive,
+              STATE_EVENT_ROOM_MEMBER,
+            ),
+          ]}
+        >
+          {/* 
+          The StoreProvider is located here to keep the example small. Normal
+          applications would located it outside of the router to establish a
+          single, global store.
+          */}
+          <StoreProvider>
+            <Card>
+              <CardContent
+                sx={{
+                  display: 'flex',
+                  justifyContent: 'start',
+                  alignItems: 'center',
+                }}
+              >
+                <label htmlFor="file-input" style={{ cursor: 'pointer' }}>
+                  <Input
+                    id="file-input"
+                    type="file"
+                    onChange={handleFileChange}
+                    sx={{ display: 'none' }}
+                  />
+                  <Button variant="contained" component="span">
+                    choose an image
+                  </Button>
+                </label>
+                {selectedFile && (
+                  <Typography variant="body1" sx={{ marginLeft: 2 }}>
+                    {selectedFile?.name}
+                  </Typography>
+                )}
+                {errorMessage && (
+                  <Typography
+                    color="error"
+                    variant="body2"
+                    sx={{ marginLeft: 2 }}
+                  >
+                    {errorMessage}
+                  </Typography>
+                )}
+              </CardContent>
+            </Card>
+            <Grid item xs={12} mt={1}>
+              <Button
+                onClick={handleFileUpload}
+                variant="contained"
+                disabled={loading}
+                fullWidth
+              >
+                Upload image
+              </Button>
+            </Grid>
+            {images.length > 0 && (
+              <ImagesList
+                key={images.length}
+                imagesData={images}
+                setImages={setImages}
+              />
+            )}
+          </StoreProvider>
+        </MuiCapabilitiesGuard>
+      </Box>
+    </>
+  );
+};
+
+export const mxcUrlToHttp = (mxcUrl: string): string | null => {
+  const match = mxcUrl.match(/^mxc:\/\/([^/]+)\/([^/]+)$/);
+
+  if (match) {
+    const serverName = match[1];
+    const mediaId = match[2];
+    return `https://${serverName}/_matrix/media/r0/download/${serverName}/${mediaId}`;
```

Comment:
this is not the correct way to translate mxc to http - we need the base HS url here.

Reply:
this was resolved by introducing the `$matrix_base_url` widget parameter in the `api` package